client/me/purchases/billing-history/utils.jsx

```diff
@@ -82,6 +83,133 @@ export function renderTransactionAmount( transaction, { translate, addingTax = f
+			'Purchase of %(quantity)s new mailboxes',
+			{
+				args: { quantity: new_quantity },
+				count: new_quantity,
+				context: 'Purchase description with additional number of mailboxes purchased',
+				comment: '%(quantity)s is additional number of mailboxes purchased',
+			}
+		);
+	}
+
+	return translate( 'Purchase of %(quantity)s mailbox', 'Purchase of %(quantity)s mailboxes', {
+		args: { quantity: licensed_quantity },
+		count: licensed_quantity,
+		context: 'Purchase description with number of mailboxes purchased',
+		comment: '%(quantity)s is number of mailboxes purchased',
+	} );
+}
+
+function renderTransactionQuantitySummaryForGoogleWorkspace(
+	licensed_quantity,
+	new_quantity,
+	isRenewal,
+	isUpgrade,
+	translate
+) {
+	if ( isRenewal ) {
+		return translate( 'Renewal for %(quantity)s user', 'Renewal for %(quantity)s users', {
+			args: { quantity: licensed_quantity },
+			count: licensed_quantity,
+			context: 'Renewal description with number of users renewed',
+			comment: '%(quantity)s is number of users renewed',
+		} );
+	}
+
+	if ( isUpgrade ) {
+		return translate( 'Purchase of %(quantity)s new user', 'Purchase of %(quantity)s new users', {
+			args: { quantity: new_quantity },
+			count: new_quantity,
+			context: 'Purchase description with additional number of users purchased',
+			comment: '%(quantity)s is additional number of users purchased',
+		} );
+	}
+
+	return translate( 'Purchase of %(quantity)s user', 'Purchase of %(quantity)s users', {
+		args: { quantity: licensed_quantity },
+		count: licensed_quantity,
+		context: 'Purchase description with number of users purchased',
+		comment: '%(quantity)s is number of users purchased',
+	} );
+}
+
+export function renderTransactionQuantitySummary(
+	{ licensed_quantity, new_quantity, type, wpcom_product_slug },
+	translate
+) {
+	if ( ! licensed_quantity ) {
+		return null;
+	}
+
+	licensed_quantity = parseInt( licensed_quantity );
+	new_quantity = parseInt( new_quantity );
+	const product = { product_slug: wpcom_product_slug };
+	const isRenewal = 'recurring' === type;
+	const isUpgrade = 'new purchase' === type && new_quantity > 0;
+
+	if ( isTitanMail( product ) ) {
+		return renderTransactionQuantitySummaryForTitanMail(
+			licensed_quantity,
+			new_quantity,
+			isRenewal,
+			isUpgrade,
+			translate
+		);
+	} else if ( isGoogleWorkspace( product ) ) {
+		return renderTransactionQuantitySummaryForGoogleWorkspace(
+			licensed_quantity,
+			new_quantity,
+			isRenewal,
+			isUpgrade,
+			translate
+		);
+	}
+	if ( isRenewal ) {
+		return translate( 'Renewal for %(quantity)s item', 'Renewal for %(quantity)s items', {
+			args: { quantity: licensed_quantity },
+			count: licensed_quantity,
+			context: 'Renewal description with number of items renewed',
+			comment: '%(quantity)s is number of items renewed',
+		} );
+	}
+
+	if ( isUpgrade ) {
+		return translate( 'Purchase of %(quantity)s new item', 'Purchase of %(quantity)s new items', {
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?